vercel-api/src/components/charts/ChartBase.ts

```
@@ -0,0 +1,510 @@
+      let key: string;
+      const date = new Date(item.timestamp);
+      
+      switch (period) {
+        case 'hour':
+          key = `${date.getFullYear()}-${date.getMonth()}-${date.getDate()}-${date.getHours()}`;
+          break;
+        case 'day':
+          key = `${date.getFullYear()}-${date.getMonth()}-${date.getDate()}`;
+          break;
+        case 'week':
+          const weekStart = new Date(date.setDate(date.getDate() - date.getDay()));
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)